### DISCUSSION TITLE
Is there any example for fluid dynamic simulation?

### DISCUSSION THREAD

**Discussion by zhang-qiang-github** [usnistgov/fipy#893] [General]
I want to simulate the blood flow in a 3D vessel. Input a 3D vessel surface, and the velocity in the boundary. I want to obtain the pressure in the surface. Is there any example to guide me?

Any suggestion is appreciated~~~

**Comment by guyer** [comment]
We do have a [moving lid Stokes flow example](https://www.ctcms.nist.gov/fipy/examples/flow/generated/examples.flow.stokesCavity.html). 

I also have an [attempt](https://github.com/guyer/phasefieldbenchmark-5) to implement a Stokes flow problem in a tube, with and without an obstruction, as described in [this benchmark description](https://pages.nist.gov/pfhub/benchmarks/benchmark5-hackathon.ipynb/). At a minimum, there are problems with the outflow condition in my implementation.

Finally, flow is incorporated in our [reactive wetting examples](https://www.ctcms.nist.gov/fipy/examples/reactiveWetting). There's a lot else going on in those examples. Hopefully @wd15 can confirm what flow regime they're in; I think it's full Navier-Stokes, but I'm not a flow guy.

If you actually want to model the pressure in the vessel wall, then the multi-phase reactive wetting example is probably what you'll want to work from. If, on the other hand, the vessel wall is just a boundary condition, then it should not be a problem to extract the pressure at that boundary from the Stokes example.

As to describing the 3D vessel surface, you'll want to do that with [Gmsh](https://gmsh.info/) and then import the domains and boundaries of interest as described for [`Gmsh2D`](https://www.ctcms.nist.gov/fipy/fipy/generated/fipy.meshes.html#fipy.meshes.Gmsh2D). While that example is in 2D, it works the same way for `Gmsh3D`.
